docusaurus/docs/iOS/client/extra-data.md

```diff
@@ -0,0 +1,212 @@
+---
+title: Extra Data
+---
+
+Extra Data is additional information that can be added to the default data of Stream. It is a dictionary of key-value pairs that can be attached to messages, users, channels, and pretty much almost every domain model in the Stream SDK.
+
+On iOS, the Extra Data is represented by the following dictionary, `[String: RawJSON]`. The `RawJSON` is an enum that can be represented by different types of values. It can be a String, Number, Boolean, Array, Dictionary, or null. In the end, this is to make the dictionary strongly typed so that it is more safe and easier to use. The code snippet below shows the simplified implementation of `RawJSON`.
+
+```swift
+indirect enum RawJSON: Codable, Hashable {
+    case number(Double)
+    case string(String)
+    case bool(Bool)
+    case dictionary([String: RawJSON])
+    case array([RawJSON])
+}
+```
+
+## Adding Extra Data
+
+Adding extra data can be done through the Server-Side SDKs or through the client SDKs. In the iOS Stream Chat SDK, you can add extra data when creating/updating a message, user channel, or any other model through our controllers.
+
+As a simple example, let's see how you can add a new email field to the current logged user.
+
+```swift
+let currentUserController = client.currentUserController()
+currentUserController.updateUserData(
+    name: "John Doe",
+    imageURL: nil,
+    userExtraData: ["email": .string("john.doe@example.com")],
+    completion: nil
+)
+```
+
+For a more complete example now, let's imagine you want to add a ticket information to a message.
+
+```swift
+let extraData: [String: RawJSON] = [
+    "ticket": .dictionary([
+        "name": .string("Rock Concert"),
+        "price": .double(20)
+    ])
+]
+let channelController = client.channelController(for: yourChannelId)
+channelController.createNewMessage(text: "A new message", extraData: extraData)
+```
+
+:::tip
+Since `RawJSON` implements `ExpressibleByDictionaryLiteral`, you can simplify the example above like this:
+```swift
+let extraData: [String: RawJSON] = [
+    "ticket": [
+        "name": .string("Rock Concert"),
+        "price": .double(20)
+    ])
+]
+```
+***Note:*** This is only available on 4.19.0+ versions of the SDK.
+:::
+
+## Reading Extra Data
+
+All of the most important domain models in the SDK have a `extraData` property that you can read the additional information that is added.
+
+Since the 4.19.0 version of the SDK you can read extra data properties very easily. The following code snippet shows how to get an email from a user's extra data.
```

Comment:
```suggestion
Since the 4.19.0 version of the SDK, you can read extra data properties very easily. The following code snippet shows how to get an email from a user's extra data.
```